crane/container.go

```diff
@@ -122,7 +122,9 @@ func (c *container) Dependencies() *Dependencies {
 	dependencies := &Dependencies{}
 	for _, link := range c.RunParams.Link() {
 		linkParts = strings.Split(link, ":")
-		dependencies.All = append(dependencies.All, linkParts[0])
+		if !dependencies.includesAsKind(linkParts[0], "link") {
```

Comment:
While you are it, I think you should rather follow the same pattern as below for VolumesFrom, to avoid having duplicates anywhere - namely:
```
		name = linkParts[0]
		if !dependencies.includes(name) {
			dependencies.All = append(dependencies.All, name)
			dependencies.Link = append(dependencies.Link, name)
		}
```